packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.test.tsx

```diff
@@ -94,6 +97,55 @@ describe('routes/Subscriptions/PaymentUpdateFormV2', () => {
     expect(screen.queryByTestId('paymentForm')).toBeInTheDocument();
   });
 
+  it('renders correctly for paypal', async () => {
+    render(<Subject customer={{ ...CUSTOMER, payment_provider: 'paypal' }} />);
+    expect(
+      screen.getByTestId('change-payment-update-button')?.getAttribute('href')
+    ).toEqual('https://www.paypal.com/myaccount/autopay/connect/ba-131243');
+  });
+
+  it('renders correctly for missing billing agreement for paypal', async () => {
+    render(
+      <Subject
+        customer={{
+          ...CUSTOMER,
+          payment_provider: 'paypal',
+          paypal_payment_error: 'missing_agreement',
+        }}
+      />
+    );
+    expect(
+      screen.queryAllByTestId('reveal-payment-modal-button').length
+    ).toEqual(2);
+
+    await act(async () => {
+      fireEvent.click(
+        screen.queryAllByTestId('reveal-payment-modal-button')[0]
+      );
+    });
+
+    await waitForExpect(() =>
+      expect(screen.queryByTestId('billing-info-modal')).toBeInTheDocument()
+    );
+  });
+
+  it('renders correctly for incorrect funding source for paypal', async () => {
+    render(
+      <Subject
+        customer={{
+          ...CUSTOMER,
+          payment_provider: 'paypal',
+          paypal_payment_error: 'funding_source',
+        }}
+      />
+    );
+    expect(
+      screen
+        .queryAllByTestId('manage-payment-update-button')[0]
```

Comment:
Does this imply that there are also multiple `manage-payment-update-button` buttons?  Why is that?

Reply:
notication and view